tests/validation/tests/rke/common.py

```diff
@@ -146,6 +148,152 @@ def wait_for_etcd_cluster_health(node, etcd_private_ip=False):
+
+
+def verify_dns_addon_images(k8sversion, rke_client, kubectl, namespace,
+                            selector):
+
+    coredns = get_component_version(rke_client, k8sversion,
+                                    "rancher/coredns-coredns")
+    # Sleep to allow the components to get to running state
+    time.sleep(5)
+    verify_component_status_with_kubectl(kubectl, namespace, selector, coredns)
+
+
+def verify_networking_addon_images(k8sversion, rke_client, kubectl,
+                                   namespace, selector):
+
+    flannel = get_component_version(rke_client, k8sversion,
+                                    "rancher/coreos-flannel")
+
+    calico = get_component_version(rke_client, k8sversion,
+                                   "rancher/calico-node")
+    # Sleep to allow the components to get to running state
+    time.sleep(5)
+    verify_component_status_with_kubectl(kubectl, namespace, selector, calico,
+                                         flannel)
+
+
+def verify_component_status_with_kubectl(kubectl, namespace, selector, *args):
+    # Method to verify addon status and images
+
+    command = "get pod --namespace " + namespace + " -l " + selector
+
+    res = kubectl.execute_kubectl_cmd(command, json_out=True)
+    result = json.loads(res)
+    timeout = 180
+    start = time.time()
+    # Check if the pod is running
+    for pod in result["items"]:
+        podstatus = pod["status"]["phase"]
+        podname = pod["metadata"]["name"]
+        print("Pod name is")
+        print(podname)
+        podreloadcommand = "get pod " + podname + " --namespace " + namespace
+        while (podstatus != "Running"):
+            if time.time() - start > timeout:
+                raise AssertionError("Timed out waiting for deletion")
+            time.sleep(.5)
+            podresult = kubectl.execute_kubectl_cmd(podreloadcommand,
+                                                    json_out=True)
+            podresult = json.loads(podresult)
+            podname = podresult["metadata"]["name"]
+            print("Pod name is")
+            print(podname)
+            podstatus = podresult["status"]["phase"]
+            print("Pod status is: ")
+            print(podstatus)
+            print("Timeout")
+            print(time.time() - start)
+        assert True
+
+    # Verify the component images
+    updatedresult = kubectl.execute_kubectl_cmd(command, json_out=True)
+    updatedresult = json.loads(updatedresult)
+    for pod in updatedresult["items"]:
+        print("Required Resource Image: ")
+        print(args[0])
+        for i in range(0, len(args)):
+            print(pod["status"]["containerStatuses"][i]["image"])
+            assert pod["status"]["containerStatuses"][i]["image"] == args[i]
+
+
+def get_system_images(rke_client, k8sversion):
+
+    # Method to obtain the system images for a k8s version  from rke cli
+
+    command = ("rke config --system-images --version " + k8sversion)
```

Comment:
This method is called from `get_component_version` which is called many times and will call `rke config` many times -- in this method `get_system_images` you should get the images from the common dictionary you populate with `build_expectedimages_dict`

Reply:
Used  global systemimagesdict to store the required images list for a specific K8s version.